grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/SumMapReduce.java

```diff
@@ -18,82 +18,51 @@
 
 package ai.grakn.graql.internal.analytics;
 
-import ai.grakn.concept.ResourceType;
 import ai.grakn.concept.TypeName;
 import ai.grakn.util.Schema;
-import org.apache.tinkerpop.gremlin.process.computer.KeyValue;
 import org.apache.tinkerpop.gremlin.structure.Vertex;
 import org.apache.tinkerpop.gremlin.util.iterator.IteratorUtils;
 
 import java.io.Serializable;
-import java.util.HashMap;
 import java.util.Iterator;
-import java.util.Map;
 import java.util.Set;
 
 public class SumMapReduce extends GraknMapReduce<Number> {
 
-    public static final String MEMORY_KEY = "sum";
-    private static final String RESOURCE_DATA_TYPE_KEY = "RESOURCE_DATA_TYPE_KEY";
-
     public SumMapReduce() {
```

Comment:
I don't know much about this, but there's a comment on the `GraknMapReduce` constructor stating it is necessary. The subclasses don't have such a comment, which might be the wrong way round.